processor/src/test/resources/fixtures/org/mapstruct/ap/test/nestedbeans/mixed/FishTankMapperImpl.java

```diff
@@ -0,0 +1,358 @@
+        }
+
+        WaterQualityOrganisationDto waterQualityOrganisationDto = new WaterQualityOrganisationDto();
+
+        waterQualityOrganisationDto.setName( waterQualityReport.getOrganisationName() );
+
+        return waterQualityOrganisationDto;
+    }
+
+    protected WaterQualityReportDto waterQualityReportToWaterQualityReportDto(WaterQualityReport waterQualityReport) {
+        if ( waterQualityReport == null ) {
+            return null;
+        }
+
+        WaterQualityReportDto waterQualityReportDto = new WaterQualityReportDto();
+
+        waterQualityReportDto.setOrganisation( waterQualityReportToWaterQualityOrganisationDto( waterQualityReport ) );
+        waterQualityReportDto.setVerdict( waterQualityReport.getVerdict() );
+
+        return waterQualityReportDto;
+    }
+
+    protected WaterQualityDto waterQualityToWaterQualityDto(WaterQuality waterQuality) {
+        if ( waterQuality == null ) {
+            return null;
+        }
+
+        WaterQualityDto waterQualityDto = new WaterQualityDto();
+
+        waterQualityDto.setReport( waterQualityReportToWaterQualityReportDto( waterQuality.getReport() ) );
+
+        return waterQualityDto;
+    }
+
+    private Ornament sourceInteriorOrnament(FishTank fishTank) {
+
+        if ( fishTank == null ) {
+            return null;
+        }
+        Interior interior = fishTank.getInterior();
+        if ( interior == null ) {
+            return null;
+        }
+        Ornament ornament = interior.getOrnament();
+        if ( ornament == null ) {
+            return null;
+        }
+        return ornament;
+    }
+
+    protected OrnamentDto ornamentToOrnamentDto(Ornament ornament) {
+        if ( ornament == null ) {
+            return null;
+        }
+
+        OrnamentDto ornamentDto = new OrnamentDto();
+
+        ornamentDto.setType( ornament.getType() );
+
+        return ornamentDto;
+    }
+
+    protected WaterPlantDto waterPlantToWaterPlantDto(WaterPlant waterPlant) {
+        if ( waterPlant == null ) {
+            return null;
+        }
+
+        WaterPlantDto waterPlantDto = new WaterPlantDto();
+
+        waterPlantDto.setKind( waterPlant.getKind() );
+
+        return waterPlantDto;
+    }
+
+    private Ornament sourceInteriorOrnament1(FishTank fishTank) {
+
+        if ( fishTank == null ) {
+            return null;
+        }
+        Interior interior = fishTank.getInterior();
+        if ( interior == null ) {
+            return null;
+        }
+        Ornament ornament = interior.getOrnament();
+        if ( ornament == null ) {
+            return null;
+        }
+        return ornament;
+    }
+
+    protected Fish fishDtoToFish(FishDto fishDto) {
+        if ( fishDto == null ) {
+            return null;
+        }
+
+        Fish fish = new Fish();
+
+        fish.setType( fishDto.getKind() );
+
+        return fish;
+    }
+
+    private String waterQualityReportDtoOrganisationName(WaterQualityReportDto waterQualityReportDto) {
```

Comment:
This method is generated by the `NestedPropertyMappingMethod` which are used for nested source properties. The names of this methods are created by joining all the element names from the `SourceReference`. Have a look at the [PropertyMapping#L487](https://github.com/mapstruct/mapstruct/blob/master/processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java#L487). The forged method has a source `waterQualityReportDto` and the mapping:
```java
@Mapping(target = "organisation.name", source = "organisationName")
```

Therefore the name is like it is. If you think that it needs to be different, I say we update #1082